packages/protocol/contracts/L1/verifiers/PseZkVerifier.sol

```diff
@@ -77,20 +118,24 @@ contract PseZkVerifier is EssentialContract, IVerifier {
 
     function calcInstance(
         address prover,
+        bytes32 blobHash,
+        bytes32 txListHash,
         TaikoData.BlockEvidence memory evidence
     )
         public
         pure
         returns (bytes32 instance)
     {
         return keccak256(
-            abi.encode(
+            abi.encodePacked(
                 evidence.metaHash,
                 evidence.parentHash,
                 evidence.blockHash,
                 evidence.signalRoot,
                 evidence.graffiti,
-                prover
+                prover,
+                blobHash,
+                txListHash
```

Comment:
Will also need to contain the x and y points so we have them in the circuits.

Reply:
The x value is the instance itself, the y value cannot be calculated without x, which is the instance. so y should not be used to calculate the instance.

Reply:
See:

```solidity
Lib4844.evaluatePoint({
        blobHash: blobHash,
        x: uint256(instance) % Lib4844.BLS_MODULUS,
        y: pf.pointValue,
        commitment: pf.pointCommitment,
        proof: pf.pointProof
});
```

Reply:
`x` is the instance after the modulus so some extra work in the circuit needed to calculate it, but shouldn't complicate things too much (hopefully).

We do really need `y` as part of the instance though so because the circuit needs to have access to it, that's why we calculate it in the smart contract so that we know the correct value.

I would propose only hashing the required data for the random point `x`: `keccak(blobHash, txListHash)`.

Reply:
Modified as suggested.